### PR TITLE
Support multiple precisions when checking touchesWith

### LINKED ISSUE #67: touchesWith() does not take precision into consideration
Seems that `touchesWith()` will always return true if the compared periods are in the same day, even if the times are not really touched and the precision is SECOND.

Example:

```
$a = Period::make('2018-01-01 06:30:00', '2018-01-01 07:30:00');
$b = Period::make('2018-01-01 09:00:00', '2018-01-01 10:00:00');
$overlap = $a->touchesWith($b);
(true)
```

This leads to `gap()` function not detecting the [gap](https://github.com/spatie/period/blob/master/src/Period.php#L283) for such cases.

### PR DESCRIPTION
**Pull request by ahallhognason**
Closes #67 

I have updated the tests to add a check with `Precision::SECOND` and confirmed that the updated `touchesWith` method now supports multiple precisions (seconds, days, etc).

### CODE REVIEW

**Comment by ahallhognason** [comment]
@brendt I extended the existing test for determining if two periods touched. I hope that's alright.

File: src/Period.php [hunk 1]
```diff
@@ -517,7 +532,7 @@ protected function createDateInterval(int $precision): DateInterval
             Precision::HOUR => 'PT1H',
             Precision::DAY => 'P1D',
             Precision::MONTH => 'P1M',
-            Precision::YEAR => 'P1Y',
```
Comment (brendt) [inline src/Period.php:520]: Could you undo these styling changes?

File: src/Visualizer.php [hunk 2]
```diff
@@ -56,7 +56,7 @@ public function visualize(array $blocks): string
         foreach ($matrix as $name => $row) {
             $lines[] = vsprintf('%s    %s', [
                 str_pad($name, $nameLength, ' '),
-                $this->toBars($row),
```
Comment (brendt) [inline src/Visualizer.php:59]: Could you undo these styling changes?

File: tests/PeriodCollectionTest.php [hunk 3]
```diff
@@ -138,7 +138,7 @@ public function no_gaps_when_periods_fully_overlap_and_end_excluded()
         $collection = new PeriodCollection(
             Period::make('2018-01-01', '2018-01-05', null, Boundaries::EXCLUDE_END),
             Period::make('2018-01-01', '2018-01-05', null, Boundaries::EXCLUDE_END),
-            Period::make('2018-01-01', '2018-01-05', null, Boundaries::EXCLUDE_END),
```
Comment (brendt) [inline tests/PeriodCollectionTest.php:141]: Could you undo these styling changes?

File: tests/PeriodTest.php [hunk 4]
```diff
@@ -92,7 +108,7 @@ public function overlappingDates(): array
              * A    [====]
              * B    [====]
              */
-            [Period::make('2018-01-01', '2018-02-01'), Period::make('2018-01-01', '2018-02-01')],
```
Comment (brendt) [inline tests/PeriodTest.php:95]: Could you undo these styling changes?

File: tests/PeriodTest.php [hunk 5]
```diff
@@ -109,7 +125,7 @@ public function noOverlappingDates()
              * A          [===]
              * B    [===]
              */
-            [Period::make('2018-02-01', '2018-02-28'), Period::make('2018-01-01', '2018-01-31')],
```
Comment (brendt) [inline tests/PeriodTest.php:112]: Could you undo these styling changes?

File: tests/PeriodTest.php [hunk 6]
```diff
@@ -546,7 +562,7 @@ public function expectedPeriodLengths()
             [13, Period::make('2018-01-01', '2018-01-15', null, Boundaries::EXCLUDE_ALL)],
 
             [24, Period::make('2018-01-01 00:00:00', '2018-01-01 23:59:59', Precision::HOUR)],
-            [24, Period::make('2018-01-01 00:00:00', '2018-01-02 00:00:00', Precision::HOUR, Boundaries::EXCLUDE_END)],
```
Comment (brendt) [inline tests/PeriodTest.php:549]: Could you undo these styling changes?

File: tests/PrecisionTest.php [hunk 7]
```diff
@@ -45,7 +45,7 @@ public function roundingDates(): array
             [Precision::DAY, '2018-02-05 00:00:00', '2018-03-05 00:00:00'],
             [Precision::HOUR, '2018-02-05 11:00:00', '2018-03-05 11:00:00'],
             [Precision::MINUTE, '2018-02-05 11:11:00', '2018-03-05 11:11:00'],
-            [Precision::SECOND, '2018-02-05 11:11:11', '2018-03-05 11:11:11'],
```
Comment (brendt) [inline tests/PrecisionTest.php:48]: Could you undo these styling changes?

**Review by brendt — CHANGES_REQUESTED** [review]
Looks good to me, but please remove all styling changes and keep the trailing commas.

**Comment by ahallhognason** [comment]
> Looks good to me, but please remove all styling changes and keep the trailing commas.

I removed the last commit that removed the trailing commas. It might be worth dropping support for PHP 7.1 and 7.2 if you want to keep trailing commas, since those versions of PHP will break on this package.

File: tests/PrecisionTest.php (hunk 7 above)
Comment (ahallhognason) [reply tests/PrecisionTest.php]: Rolled back last commit to undo changes.

File: tests/PeriodTest.php (hunk 6 above)
Comment (ahallhognason) [reply tests/PeriodTest.php]: Rolled back last commit to undo changes.

File: tests/PeriodTest.php (hunk 5 above)
Comment (ahallhognason) [reply tests/PeriodTest.php]: Rolled back last commit to undo changes.

File: tests/PeriodTest.php (hunk 4 above)
Comment (ahallhognason) [reply tests/PeriodTest.php]: Rolled back last commit to undo changes.

File: tests/PeriodCollectionTest.php (hunk 3 above)
Comment (ahallhognason) [reply tests/PeriodCollectionTest.php]: Rolled back last commit to undo changes.

File: src/Visualizer.php (hunk 2 above)
Comment (ahallhognason) [reply src/Visualizer.php]: Rolled back last commit to undo changes.

File: src/Period.php (hunk 1 above)
Comment (ahallhognason) [reply src/Period.php]: Rolled back last commit to undo changes.

**Comment by brendt** [comment]
I'll fix that on master, thanks!

**Comment by brendt** [comment]
Fixed with 1.5.1: https://github.com/spatie/period/releases/tag/1.5.1

**Comment by ahallhognason** [comment]
Would you mind changing the label to `hacktoberfest-accepted`? It looks like they are quite particular on the label.

**Comment by freekmurze** [comment]
There you go!

**Pull request merged**